packages/react/src/components/Toggle/Toggle.tsx

```diff
@@ -1,16 +1,96 @@
 /**
- * Copyright IBM Corp. 2021
+ * Copyright IBM Corp. 2021, 2023
  *
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useRef } from 'react';
+import React, { MouseEventHandler, useRef } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { useControllableState } from '../../internal/useControllableState';
 import { usePrefix } from '../../internal/usePrefix';
 
+type ExcludedAttributes =
+  | 'aria-labelledby'
+  | 'aria-checked'
+  | 'type'
+  | 'role'
+  | 'id'
+  | 'size'
+  | 'onClick';
+
+export interface ToggleProps
+  extends Omit<
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    ExcludedAttributes
+  > {
+  /**
+   * Specify another element's id to be used as the label for this toggle
+   */
+  'aria-labelledby'?: string;
+
+  /**
+   * Provide an id that unique represents the underlying `<button>`
+   */
+
+  id: string;
+
+  /**
+   * Specify the label for the "off" position
+   */
+  labelA?: string | undefined;
+
+  /**
+   * Specify the label for the "on" position
+   */
+  labelB?: string | undefined;
+
+  /**
+   * Provide the text that will be read by a screen reader when visiting this
+   * control. This should be provided unless 'aria-labelledby' is set instead
+   * or you use an external <label> element with its "for" attribute set to the
+   * toggle's id.
+   */
+  labelText?: string | undefined;
+
+  /**
+   * If true, the side labels (props.labelA and props.labelB) will be replaced by
+   * props.labelText (if passed), so that the toggle doesn't render a top label.
+   */
+  hideLabel?: boolean;
+
+  /**
+   * Provide an event listener that is called when the control is toggled
+   */
+  onClick: MouseEventHandler<HTMLDivElement> | undefined;
```

Comment:
looks like this function also gets called when the button is activated via keyboard? in that case we need to adjust the type
```suggestion
  onClick: MouseEventHandler<HTMLDivElement> | MouseEventHandler<HTMLButtonElement> | undefined;
```

Reply:
Thanks, good catch! 

Added support typings for both mouse and keyboard events.